source/code/core/utils/public/ice/userdata.hxx

```diff
@@ -0,0 +1,91 @@
+#pragma once
+#include <ice/base.hxx>
+#include <typeinfo>
+
+namespace ice
+{
+
+    template<bool IsValidated = false>
+    struct UserdataBase
```

Comment:
This type might come in handy, but more test are required to see if we can in release builds just replace it with a `void*` (post #95

Reply:
To be seen if we need this, and what would be the impact.